tools/infrastructure/api_compare.py

```diff
@@ -0,0 +1,368 @@
+#!/usr/bin/env python
+
+import json
+import sys
+import argparse
+import xml.etree.ElementTree as ET
+from collections import namedtuple
+reload(sys)
+
+# Command line arguments parsing
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-f1", "--mobile_api_file",
+    type=str, default="MOBILE_API.xml",
+    help='Enter mobile api file name! ("MOBILE_API.xml" by default)'
+)
+
+parser.add_argument(
+    "-f2", "--hmi_api_file",
+    type=str, default="HMI_API.xml",
+    help='Enter hmi api file name! ("HMI_API.xml" by default)'
+)
+
+parser.add_argument(
+    "-o", "--output",
+    type=str, default="console",
+    choices=['xml', 'json', 'console'],
+    help='Choose report type! ("console" by default)'
+)
+
+args = parser.parse_args()
+
+# Enable Utf-8 support
+sys.setdefaultencoding('utf-8')
+
+mobile_api_tree = ET.parse(args.mobile_api_file)
+hmi_api_tree = ET.parse(args.hmi_api_file)
+
+mobile_root = mobile_api_tree.getroot()
+hmi_root = hmi_api_tree.getroot()
+
+mob_apis = {}
+hmi_apis = {}
+summary_result = {}
+
+
+# Class defining colorful output.
+# Works under UNIX ONLY as Windows does not support ANSI escape sequences
+
+class colors:
+    HEADER = '\033[95m'
+    BLUE = '\033[94m'
+    GREEN = '\033[92m'
+    WARN = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+
+Rpc = namedtuple("rpc", ["rpc_name", "rpc_type"])
+
+# Parameters which should be excluded from comparison
+exclude_redundant_from_hmi_rpc_filter = {
```

Comment:
@Please group all rules in one place. 
attributes_comparison_rules is rely far from this ones